api/lib/infrastructure/repositories/membership-repository.js

```diff
@@ -88,18 +110,17 @@ module.exports = {
       .then((memberships) => bookshelfToDomainConverter.buildDomainObjects(BookshelfMembership, memberships));
   },
 
-  updateById({ id, membershipAttributes }) {
-    return new BookshelfMembership({ id })
-      .save(membershipAttributes, { patch: true, method: 'update', require: true })
-      .then((updatedMembership) => updatedMembership.refresh({
-        withRelated: ['user', 'organization'],
-      }))
-      .then(
-        (membership) => bookshelfToDomainConverter.buildDomainObject(BookshelfMembership, membership),
-        (err) => {
-          throw new MembershipUpdateError(err.message);
-        },
-      );
+  async updateById({ id, membership }, domainTransaction = DomainTransaction.emptyTransaction()) {
+    let updatedMembership;
+    try {
+      updatedMembership = await new BookshelfMembership({ id })
+        .save(membership, { patch: true, method: 'update', require: true, transacting: domainTransaction.knexTransaction });
+    } catch (err) {
+      throw new MembershipUpdateError(err.message);
+    }
+
+    updatedMembership = await updatedMembership.refresh({ withRelated: ['user', 'organization'], transacting: domainTransaction.knexTransaction });
```

Comment:
**suggestion**
La doc knex parle de `refresh` comme de 
* `Update the attributes of a model`
*  `// Fetch a relation that has not been eager loaded yet`

C'est clair, mais ça va contre l'immutabilité par défaut. Je propose donc
```suggestion
    const newMembershipWithUserAndOrganization = await updatedMembership.refresh({ withRelated: ['user', 'organization'], transacting: domainTransaction.knexTransaction });
```
ou
```suggestion
    const newMembershipWithRelated = await updatedMembership.refresh({ withRelated: ['user', 'organization'], transacting: domainTransaction.knexTransaction });
```

Reply:
Vu ensemble : `const updatedMembershipWithUserAndOrganization`